llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -1410,6 +1410,12 @@ AArch64TargetLowering::AArch64TargetLowering(const TargetMachine &TM,
     }
   }
 
+  for (MVT VT : {MVT::v8i16, MVT::v4i32}) {
+    setOperationAction(ISD::TRUNCATE_SSAT_S, VT, Custom);
```

Comment:
If this uses isTypeDesirableForOp to only generate the truncates of the expected size, then these can be legal as opposed to custom. (If they were custom then there would need to be something in LowerOperation to handle them).

Reply:
agree. I also added a comment with that intention and I'm thinking maybe adding code to `DAGCombiner` can solve it. 

do you think I should include this in this patch as well?

Reply:
Sorry - I'm not sure I understand what you mean. I believe in this version of the code just making them Legal should be fine, but I'm not sure which comment this was referring to or what code to DAG combiner.

Reply:
never mind. I was left comment at wrong place